packages/forms-web-app/src/controllers/common/enter-code.js

```diff
@@ -183,21 +197,37 @@ const postEnterCode = (views, appealInSession) => {
 			});
 		}
 
+		const enrolUsersFlag = await isFeatureActive(FLAG.ENROL_USERS);
 		const isReturningUser = req.session.newOrSavedAppeal === 'return';
 
-		let tokenValidResult = async () => {
+		/** @type {import('appeals-service-api').Api.AppealUser|undefined} */
+		let user;
+		/** @type {string|undefined} */
+		let sessionEmail;
+
+		if (enrolUsersFlag) {
+			sessionEmail = getSessionEmail(req.session, appealInSession);
+			user = await apiClient.getUserByEmailV2(sessionEmail);
```

Comment:
I _think_ this will throw if not found? - if it does I guess we don't need need the throw below.

Your code in [email-address.js](https://github.com/Planning-Inspectorate/appeal-planning-decision/pull/3374/files#diff-19919ab435586cdef03ebcb0bf97cd75e4d28099beb4332cb9f7edf9b1d0876b) implies it does.

Reply:
yes both can throw